jaeger-core/src/test/java/com/uber/jaeger/reporters/RemoteReporterTest.java

```diff
@@ -213,15 +213,18 @@ public void testFlushWhenQueueFull() {
   public void testFlushUpdatesQueueLength() throws Exception {
     // change sender to blocking mode
     sender.permitAppend(0);
-    RemoteReporter remoteReporter = (RemoteReporter) reporter;
-    remoteReporter.flush();
+
+    await("flush() execution via TimerTask").atMost(5, TimeUnit.SECONDS).until(
+        () -> metricsReporter.gauges.get("jaeger.reporter-queue") != null
+    );
```

Comment:
TBH, I don't know why we're executing flush immediately
```java
    flushTimer.schedule(
        new TimerTask() {
          @Override
          public void run() {
            flush();
          }
        },
        0,
        flushInterval);
  }
```

It's that `0` argument - I think we should remove it.